examples/conway/main.rs

```diff
@@ -0,0 +1,394 @@
+            }
+            if input.key_pressed(VirtualKeyCode::Space) {
+                // Space is frame-step, so ensure we're paused
+                paused = true;
+            }
+            if input.key_pressed(VirtualKeyCode::R) {
+                life.randomize();
+            }
+            // Handle mouse. This is a bit involved since support some simple
+            // line drawing (mostly because it makes nice looking patterns).
+            let (mouse_cell, mouse_prev_cell) = input
+                .mouse()
+                .map(|(mx, my)| {
+                    let (dx, dy) = input.mouse_diff();
+                    let prev_x = mx - dx;
+                    let prev_y = my - dy;
+                    let dpx = hidpi_factor as f32;
+                    let (w, h) = (p_width as f32 / dpx, p_height as f32 / dpx);
+                    let mx_i = ((mx / w) * (SCREEN_WIDTH as f32)).round() as isize;
+                    let my_i = ((my / h) * (SCREEN_HEIGHT as f32)).round() as isize;
+                    let px_i = ((prev_x / w) * (SCREEN_WIDTH as f32)).round() as isize;
+                    let py_i = ((prev_y / h) * (SCREEN_HEIGHT as f32)).round() as isize;
+                    ((mx_i, my_i), (px_i, py_i))
+                })
+                .unwrap_or_default();
+
+            if input.mouse_pressed(0) {
+                debug!("Mouse click at {:?}", mouse_cell);
+                draw_state = Some(life.toggle(mouse_cell.0, mouse_cell.1));
+            } else if let Some(draw_alive) = draw_state {
+                let release = input.mouse_released(0);
+                let held = input.mouse_held(0);
+                debug!("Draw at {:?} => {:?}", mouse_prev_cell, mouse_cell);
+                debug!("Mouse held {:?}, release {:?}", held, release);
+                // If they either released (finishing the drawing) or are still
+                // in the middle of drawing, keep going.
+                if release || held {
+                    debug!("Draw line of {:?}", draw_alive);
+                    life.set_line(
+                        mouse_prev_cell.0,
+                        mouse_prev_cell.1,
+                        mouse_cell.0,
+                        mouse_cell.1,
+                        draw_alive,
+                    );
+                }
+                // If they let go or are otherwise not clicking anymore, stop drawing.
+                if release || !held {
+                    debug!("Draw end");
+                    draw_state = None;
+                }
+            }
+            // Adjust high DPI factor
+            if let Some(factor) = input.hidpi_changed() {
+                hidpi_factor = factor;
+            }
+            // Resize the window
+            if let Some(size) = input.window_resized() {
+                let size = size.to_physical(hidpi_factor);
+                let width = size.width.round() as u32;
+                let height = size.height.round() as u32;
```

Comment:
nitpick, but you could just reuse the `p_width` and `p_height` captured variables, no?

Reply:
Nice catch.